lib/Service/UserMigrationService.php

```diff
@@ -124,13 +162,39 @@ public function import(string $path, ?OutputInterface $output = null): void {
 		$output->writeln("Importing from ${path}…");
 		$importSource = new ImportSource($path);
 
+		$context = $this->coordinator->getRegistrationContext();
+
 		try {
-			// TODO check versions
+			if ($context === null) {
+				throw new UserMigrationException("Failed to get context");
+			}
+			$migratorVersions = $importSource->getMigratorVersions();
+
+			if (!$this->canImport($importSource, $migratorVersions[static::class] ?? null)) {
+				throw new UserMigrationException("Version ${$migratorVersions[static::class]} for main class ".static::class." is not compatible");
+			}
+
+			// Check versions
+			foreach ($context->getUserMigrators() as $migratorRegistration) {
+				/** @var IMigrator $migrator */
+				$migrator = $this->container->get($migratorRegistration->getService());
+				if (!$migrator->canImport($importSource, $migratorVersions[get_class($migrator)] ?? null)) {
+					throw new UserMigrationException("Version ".($migratorVersions[get_class($migrator)] ?? 'null')." for migrator ".get_class($migrator)." is not compatible");
+				}
+			}
 
 			$user = $this->importUser($importSource, $output);
 			$this->importAccountInformation($user, $importSource, $output);
 			$this->importAppsSettings($user, $importSource, $output);
 			$this->importFiles($user, $importSource, $output);
+
+			// Run imports of registered migrators
+			foreach ($context->getUserMigrators() as $migratorRegistration) {
+				/** @var IMigrator $migrator */
+				$migrator = $this->container->get($migratorRegistration->getService());
+				$migrator->import($user, $importSource, $output, $migratorVersions[get_class($migrator)] ?? null);
```

Comment:
```suggestion
				$migrator->import($user, $importSource, $output, $migratorVersions[get_class($migrator)]);
```
Would this be the same @come-nc ?